### ISSUE TITLE
[Feature]: Get logits instead of lobprobs for distillation

### ISSUE THREAD

**Issue by nivibilla** [vllm-project/vllm#8926]
### 🚀 The feature, motivation and pitch

Im trying to do offline distillation by using the logits from a teacher model. And i would like to use vllm to generate these logits as it is much faster than doing it via transformers.

### Alternatives

Here is the huggingface transformers equivalent.

```
# Load teacher model
teacher_model = AutoModelForCausalLM.from_pretrained(config["models"]["teacher"])
teacher_model = teacher_model.to(device)
teacher_model.eval()

def compute_logits(batch):
    inputs = teacher_tokenizer(batch["text"], return_tensors="pt", truncation=True, max_length=config["tokenizer"]["max_length"], padding="max_length").to(device)
    with torch.no_grad():
        outputs = teacher_model(**inputs)
    batch["logits"] = outputs.logits.cpu().numpy().tolist()
    return batch
```

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
Please refer to the more detailed discussion in https://github.com/vllm-project/vllm/issues/11397.
